### PR TITLE
Tests our README to prevent accidental breakages

### PR DESCRIPTION
**Pull request by LukeWood**
Making sure our quickstart doesn't accidentally get broken due to API changes is pretty important.  I've accidentally broken it a few times and figured I would add a test to cover it.

### CODE REVIEW

**Comment by bhack** [comment]
Probably it is better to parse and test the readme directly to maintain a single source of truth. 
E.g.
https://github.com/anko/txm

But probably there many other solutions to maintain a single source.

File: keras_cv/readme_quickstart_test.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+If you change this test you are responsible for creating a PR to update:
+- https://github.com/keras-team/keras-io/blob/master/templates/keras_cv/index.md
+- https://github.com/keras-team/keras-cv/blob/master/README.md#quickstart
+
+Thank you!
+"""
+
+
+@pytest.mark.skipif(
```
Comment (ianstenbit) [inline keras_cv/readme_quickstart_test.py:32]: IIRC we had decided to stop with this strategy and instead use --ignore flags in our CI testing flows to skip tests we don't want to run.

**Comment by ianstenbit** [comment]
> Probably it is better to parse and test the readme directly to maintain a single source of truth. E.g. https://github.com/anko/txm
> 
> But probably there many other solutions to maintain a single source.

I agree -- if there are ways to make this a single source that aren't too much overhead I'd prefer that. Having two copies makes this test much more fragile.

**Comment by LukeWood** [comment]
> I agree -- if there are ways to make this a single source that aren't too much overhead I'd prefer that. Having two copies makes this test much more fragile.

yeah - although no matter what we have a cross-repo duplication here too.   

Personally I think given that this also lives in keras.io theres not much of a way to fully de-duplicate this without over-engineering.

File: keras_cv/readme_quickstart_test.py (hunk 1 above)
Comment (LukeWood) [reply keras_cv/readme_quickstart_test.py:32]: it still exists several places that involve fitting models.  Lets fix it all in a batch - but for now I think this really only belongs in GCBRun.  

The tracing is pretty expensive on people's local CPUs and could be a huge painpoint for local devs.

Comment (LukeWood) [reply keras_cv/readme_quickstart_test.py:35]: lets resolve this and merge, this PR has been shelved for awhile and adds value

**Review by ianstenbit — CHANGES_REQUESTED** [review]
I'm still not thrilled about this having to be a separate bit of code from the quickstart. Is there an If-this-then-that equivalent we can use to enforce it?

Also -- what about just `quickstart_test.py`? The "README" prefix makes me want to read this file but it's not particularly relevant for most folks

**Comment by LukeWood** [comment]
@ianstenbit I renamed the test.  

I don't think this is the perfect solution -- but I do think it is better than not testing the Quickstart code at all.  It's quite bad for package impression if the quickstart fails.  

Do we have a good alternative that doesn't involve a unit test cloning keras-IO?  I'm open to an alternative, but in the absence of a good one I want to make sure we at least have coverage on this blessed workflow.

File: keras_cv/quickstart_test.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import os
+
+import pytest
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+import keras_cv
+
+"""
+ATTENTION!!!
```
Comment (fchollet) [inline keras_cv/quickstart_test.py:23]: Perhaps slightly hacky but you could open the README or quickstart md files, concatenate together the `python` code blocks, and run that.

Comment (LukeWood) [reply keras_cv/quickstart_test.py:23]: That could work -- it does still leave the keras.io problem though.

Comment (LukeWood) [reply keras_cv/quickstart_test.py:23]: Ok I've kept thinking:

maybe keras.io could unit test the KerasCV quickstart using the same coding strategy?

**Comment by LukeWood** [comment]
Closing to take a different approach.

**Pull request closed without merging**